test(lsp): `initializationOptions` test

Sysix opened the pull request:
> This PR adds comprehensive test coverage for LSP initializationOptions handling

> Implemented a custom test/configuration LSP method in the test infrastructure to introspect workspace worker configuration state

camc314 commented:
### Merge activity

* **Dec 5, 3:34 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 5, 3:34 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
test(lsp): `initializationOptions` test (#16544)

> This PR adds comprehensive test coverage for LSP initializationOptions handling

> Implemented a custom test/configuration LSP method in the test infrastructure to introspect workspace worker configuration state
